src/coap_client.c

```diff
@@ -26,6 +26,46 @@ bool golioth_client_is_connected(struct golioth_client *client)
     return client->session_connected;
 }
 
+void golioth_coap_next_token(uint8_t token[GOLIOTH_COAP_TOKEN_LEN])
+{
+    static golioth_sys_mutex_t token_mut;
+    static uint8_t stored_token[GOLIOTH_COAP_TOKEN_LEN] = {0};
+
+    if (!token_mut)
+    {
+        token_mut = golioth_sys_mutex_create();
```

Comment:
Drat... obviously you're right but that complicates things. I did it this way because it inserts a variable amount of time between power up and getting the clock cycles for the random seed.

Reply:
Ah, I should be able to initialize the mutex early, and add a variable to track whether or not random has been seeded. This protects against the race condition but still allows for a variable number of clock cycles for the seed.